docs/api/security/platform_service.md

```diff
@@ -0,0 +1,6 @@
+<h2 id="platform-service">Platfrom Service API</h2>
+
+Platform service introduces system reset and PSA Lifecycle APIs.
+PSA platform service may be extended by target vendors to expose target peripherals only accessible from secure side.
+
+Todo: link to doxygen
```

Comment:
Query: What are the use cases for this API?

Also, please add an example, in addition to the Doxygen.